### PR TITLE
[dashboard] Fix Teams & Projects top-level menu UX

### PR DESCRIPTION
**Pull request by jankeromnes**
Fixes https://github.com/gitpod-io/gitpod/issues/4622

<table><tr><th>BEFORE</th><th>AFTER</th></tr>
<tr><td>Dropdown is always split</td><td>Dropdown is only split on sub-pages</td></tr>
<tr><td>
Screenshot 2021-08-05 at 17 52 11
</td><td>
Screenshot 2021-08-05 at 17 58 20
</td></tr><tr><td>
Screenshot 2021-08-05 at 17 45 12
</td><td>
Screenshot 2021-08-05 at 17 52 01
</td></tr></table>

@gtsiolis What do you think about this change? Is it a good trade-off? (Nicer on root pages, but still allows breadcrumb navigation when there are breadcrumbs.)

EDIT: Now also fixes https://github.com/gitpod-io/gitpod/issues/5072 and fixes https://github.com/gitpod-io/gitpod/issues/5073

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **jankeromnes** after the PR has been reviewed.

Associated issue: *#4622*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by gtsiolis — APPROVED** [review]
@jankeromnes this looks certainly much better than before!

UX looks good to merge. Feel free to request a review for the frontend code if needed. 🏀 

Since we're here what do you think of also changing the hover color of the dropdown entries to `bg-gray-100`? 🤓 

Optionally, if this takes less only a few minutes to fix we could also highlight the actively selected team as described in https://github.com/gitpod-io/gitpod/issues/5072. However, feel free to leave this out of the scope of these changes. ✂️

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: cecce1b0a8049d3c9f9c9099e00829aa8d79b33d</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5077#pullrequestreview-723565500" title="Approved">gtsiolis</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **jankeromnes** after the PR has been reviewed.

Associated issue: *#4622*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by jankeromnes** [comment]
Many thanks for the quick review! ⚡ Will take a look at the proposed improvements. 😁

/approve

/hold

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5077#pullrequestreview-723565500" title="Approved">gtsiolis</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5077#issuecomment-893600869" title="Approved">jankeromnes</a>*

Associated issue: *#4622*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by jankeromnes** [comment]
> Since we're here what do you think of also changing the hover color of the dropdown entries to `bg-gray-100`? 🤓 

Is it cool to change them for all Dropdowns everywhere? 👀 (It's a much-used component!)

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Comment by gtsiolis** [comment]
>  Is it cool to change them for all Dropdowns everywhere?

Yeap! 🚢

**Comment by jankeromnes** [comment]
> Optionally, if this takes less only a few minutes to fix we could also highlight the actively selected team as described in #5072. However, feel free to leave this out of the scope of these changes. ✂️

Hmm, this isn't super easy, since the team selector uses the `ContextMenu` component, which doesn't really allow custom backgrounds yet.

However, an easy fix could be to make the `active` item have `bg-gray-50` everywhere. Would this look okay?

<table><tr><th>BEFORE</th><th>AFTER</th></tr>
<tr><td>
Screenshot 2021-08-05 at 18 50 43
</td><td>
Screenshot 2021-08-05 at 18 47 46
</td></tr></table>

(I could also quickly fix the background color "cutting into" the corners.)

**Comment by jankeromnes** [comment]
Okay, I think the "active" `bg-gray-50` background looks okay like this:

<table><tr><th>BEFORE</th><th>AFTER</th></tr><tr><td>
Screenshot 2021-08-06 at 08 57 18
</td><td>
Screenshot 2021-08-06 at 08 58 05
</td></tr><tr><td>
Screenshot 2021-08-06 at 08 57 27
</td><td>
Screenshot 2021-08-06 at 08 58 14
</td></tr><tr><td>
Screenshot 2021-08-06 at 08 57 37
</td><td>
Screenshot 2021-08-06 at 08 58 20
</td></tr><tr><td>
Screenshot 2021-08-06 at 08 57 44
</td><td>
Screenshot 2021-08-06 at 08 58 27
</td></tr></table>

In dark, we use the same `bg-gray-800` for active and hover:

<table><tr><th>BEFORE</th><th>AFTER</th></tr><tr><td>
Screenshot 2021-08-06 at 09 01 19
</td><td>
Screenshot 2021-08-06 at 09 02 00
</td></tr><tr><td>
Screenshot 2021-08-06 at 09 01 28
</td><td>
Screenshot 2021-08-06 at 09 02 08
</td></tr><tr><td>
Screenshot 2021-08-06 at 09 01 35
</td><td>
Screenshot 2021-08-06 at 09 02 14
</td></tr><tr><td>
Screenshot 2021-08-06 at 09 01 44
</td><td>
Screenshot 2021-08-06 at 09 02 20
</td></tr></table>

@gtsiolis Still looks good to you?

File: components/dashboard/src/components/ContextMenu.tsx
Comment (gtsiolis) [inline components/dashboard/src/components/ContextMenu.tsx:87]: **issue**: This is minor but there's a small gap on the corners due to corner radiuses. It should be barely noticable but if there's an easy fix let's also include it here. Otherwise, a follow up issue could suffice. 🏓

Comment (gtsiolis) [inline components/dashboard/src/components/ContextMenu.tsx:87]: **suggestion**: What do you think of using `bg-gray-800` for hover on dark theme to differentiate the active and hover states?

**Review by gtsiolis — APPROVED** [review]
Looks great @jankeromnes!

Left two minor comments. Besides these, UX looks good to merge! 🚢 

The changes here would probably also resolve https://github.com/gitpod-io/gitpod/issues/5073, right?

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 6c62fdd49576eb88680dfbc0fe8d7860f3a92c6c</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5077#pullrequestreview-724080443" title="Approved">gtsiolis</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5077#issuecomment-893600869" title="Approved">jankeromnes</a>*

Associated issue: *#4622*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by gtsiolis — APPROVED** [review]
**praise**: Oh, almost forgot! Thanks for using a BEFORE / AFTER tables here for screenshots with the changes. Makes reviewing UX changes so much better. 🌟

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5077#pullrequestreview-724088821" title="Approved">gtsiolis</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5077#issuecomment-893600869" title="Approved">jankeromnes</a>*

Associated issue: *#4622*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/components/ContextMenu.tsx
Comment (jankeromnes) [reply components/dashboard/src/components/ContextMenu.tsx:87]: Oh no 🙈 that's so unsatisfying 😬

Comment (jankeromnes) [reply components/dashboard/src/components/ContextMenu.tsx:87]: Sure!

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

File: components/dashboard/src/components/ContextMenu.tsx
Comment (jankeromnes) [reply components/dashboard/src/components/ContextMenu.tsx]: Follow-up issue: https://github.com/gitpod-io/gitpod/issues/5086

**Comment by jankeromnes** [comment]
Okay, should be good to launch as is! 🚀

Releasing the hold-down clamps:

/unhold

**Pull request merged**
